### ISSUE TITLE
ConversationActivity.show(this); Crashing

### ISSUE THREAD

**Issue by rad182** [smooch/smooch-android#24]
Gradle dependencies
```
compile 'io.smooch:core:latest.release'
compile 'io.smooch:ui:latest.release'
```

Crash
```java.lang.AssertionError: Missing field in io.smooch.core.c.i
at com.google.gson.b.a.r.<init>(TypeAdapters.java:812)
at com.google.gson.b.a.q$24.a(TypeAdapters.java:838)
at com.google.gson.f.a(Gson.java:416)
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:135)
at com.google.gson.b.a.k$1.<init>(ReflectiveTypeAdapterFactory.java:105)
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:104)
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:160)
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:96)
at com.google.gson.f.a(Gson.java:416)
at com.google.gson.b.a.b.a(CollectionTypeAdapterFactory.java:53)
at com.google.gson.f.a(Gson.java:416)
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:135)
at com.google.gson.b.a.k$1.<init>(ReflectiveTypeAdapterFactory.java:105)
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:104)
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:160)
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:96)
at com.google.gson.f.a(Gson.java:416)
at com.google.gson.f.a(Gson.java:653)
at com.google.gson.f.a(Gson.java:640)
at com.google.gson.f.b(Gson.java:595)
at com.google.gson.f.b(Gson.java:575)
at io.smooch.core.b.a.d.a(Unknown Source)
at io.smooch.core.b.g.a(Unknown Source)
at io.smooch.core.service.SmoochService.k(Unknown Source)
at io.smooch.core.q$6.run(Unknown Source)
at io.smooch.core.q.a(Unknown Source)
at io.smooch.core.q.i(Unknown Source)
at io.smooch.core.c.c(Unknown Source)
at io.smooch.ui.c.a.onResume(Unknown Source)
at android.support.v4.app.Fragment.performResume(Fragment.java:2020)
at android.support.v4.app.ag.a(FragmentManager.java:1107)
at android.support.v4.app.ag.a(FragmentManager.java:1252)
at android.support.v4.app.ag.a(FragmentManager.java:1234)
at android.support.v4.app.ag.p(FragmentManager.java:2056)
at android.support.v4.app.ac.i(FragmentController.java:196)
at android.support.v4.app.y.onResumeFragments(FragmentActivity.java:504)
at android.support.v4.app.y.onPostResume(FragmentActivity.java:493)
at android.support.v7.a.m.onPostResume(AppCompatActivity.java:178)
at android.app.Activity.performResume(Activity.java:5095)
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2565)
at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:2603)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2089)
at android.app.ActivityThread.access$600(ActivityThread.java:130)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:4745)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:511)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NoSuchFieldException: Unsent
at java.lang.Class.getField(Class.java:787)
at com.google.gson.b.a.r.<init>(TypeAdapters.java:801)
at com.google.gson.b.a.q$24.a(TypeAdapters.java:838) 
at com.google.gson.f.a(Gson.java:416) 
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:135) 
at com.google.gson.b.a.k$1.<init>(ReflectiveTypeAdapterFactory.java:105) 
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:104) 
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:160) 
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:96) 
at com.google.gson.f.a(Gson.java:416) 
at com.google.gson.b.a.b.a(CollectionTypeAdapterFactory.java:53) 
at com.google.gson.f.a(Gson.java:416) 
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:135) 
at com.google.gson.b.a.k$1.<init>(ReflectiveTypeAdapterFactory.java:105) 
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:104) 
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:160) 
at com.google.gson.b.a.k.a(ReflectiveTypeAdapterFactory.java:96) 
at com.google.gson.f.a(Gson.java:416) 
at com.google.gson.f.a(Gson.java:653) 
at com.google.gson.f.a(Gson.java:640) 
at com.google.gson.f.b(Gson.java:595) 
at com.google.gson.f.b(Gson.java:575) 
at io.smooch.core.b.a.d.a(Unknown Source) 
at io.smooch.core.b.g.a(Unknown Source) 
at io.smooch.core.service.SmoochService.k(Unknown Source) 
at io.smooch.core.q$6.run(Unknown Source) 
at io.smooch.core.q.a(Unknown Source) 
at io.smooch.core.q.i(Unknown Source) 
at io.smooch.core.c.c(Unknown Source) 
at io.smooch.ui.c.a.onResume(Unknown Source) 
at android.support.v4.app.Fragment.performResume(Fragment.java:2020) 
at android.support.v4.app.ag.a(FragmentManager.java:1107) 
at android.support.v4.app.ag.a(FragmentManager.java:1252) 
at android.support.v4.app.ag.a(FragmentManager.java:1234) 
at android.support.v4.app.ag.p(FragmentManager.java:2056) 
at android.support.v4.app.ac.i(FragmentController.java:196) 
at android.support.v4.app.y.onResumeFragments(FragmentActivity.java:504) 
at android.support.v4.app.y.onPostResume(FragmentActivity.java:493) 
at android.support.v7.a.m.onPostResume(AppCompatActivity.java:178) 
at android.app.Activity.performResume(Activity.java:5095) 
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2565) 
at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:2603) 
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2089) 
at android.app.ActivityThread.access$600(ActivityThread.java:130) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195) 
at android.os.Handler.dispatchMessage(Handler.java:99) 
at android.os.Looper.loop(Looper.java:137) 
at android.app.ActivityThread.main(ActivityThread.java:4745) 
at java.lang.reflect.Method.invokeNative(Native Method) 
at java.lang.reflect.Method.invoke(Method.java:511) 
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
at dalvik.system.NativeStart.main(Native Method)```

**Comment by jugarrit** [comment]
Is this a regression?
I am unable to reproduce the problem by simply calling `ConversationActivity.show(this);` 
Do you think you could provide more precise steps, or even a sample app?

Thanks!

**Comment by rad182** [comment]
@jugarrit I just added a button to call `ConversationActivity.show(this);` and it crashes.

**Comment by rad182** [comment]
any update for this?

**Comment by jugarrit** [comment]
Sorry! Still haven't had a chance to repro. I'll get back to you once I'm able.

**Comment by rad182** [comment]
I can email you my repo and maybe it will help you?

**Comment by jugarrit** [comment]
Sure! Send it to julian@smooch.io

**Comment by langhoangal** [comment]
You obfuscated your code, right? The log quite similar, i don't know. But it good if you can try in debug mode to see more clearly. And (i'm not sure if i remember exactly) i think i got it before, the reason related to Gson some how.

**Comment by ravikumar-n** [comment]
@jugarrit Got the same issue in production build, but when we disabled proguard and built, it works.  

**Project dependencies:**
```
 // Okhttp
 compile 'com.squareup.okhttp3:okhttp:3.2.0'

// Gson
compile 'com.google.code.gson:gson:2.6.2'

// Smooch - Live chat
compile 'io.smooch:core:3.2.1'
compile 'io.smooch:ui:3.2.1'
compile 'com.google.android.gms:play-services-gcm:9.0.2'
```
The following proguard rules also doesn't help in resolving the issue:
```
## Smooch
-dontwarn okio.**
-keep class okio.**
-keep class com.google.gson.**
```
**Device Info** :
 Dell Venue-7, Api-4.4.4

**Comment by sreejithileaf** [comment]
Hi Julian,
The sample app in the git hub is working fine with Android Studio, but i'm facing some kind of issues while converting it as an eclipse project. The two things that i noticed:
1) Used the latest jars ( core-3.2.1.jar, ui-3.2.1.jar) and run the app in a device. When called this line ConversationActivity.show(this). app crashed!

2) Used some old versions (core-2.3.1.jar, ui-2.3.1.jar). This time crash not occurred but Showing an error message 'Could not connect to server.' Unable to send any messages.

**Comment by skolarianer** [comment]
@ravikumar-n: Try the following rules...

```
-dontwarn okio.**
-keep class com.google.gson.** { *; }
-keepclassmembers enum * { *; }
-keepattributes Signature
-keepattributes *Annotation*
```

**Comment by jugarrit** [comment]
I will be adding proguard rules in the next release that could address this problem. I'll be closing this issue at that time and will re-open if it continues to occur.

**Comment by jugarrit** [comment]
Just pushed a proguard fix in 4.0.5. Give it a shot!
